rand_distr/src/normal_inverse_gaussian.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{Distribution, Float, InverseGaussian, Standard, StandardNormal};
+use rand::Rng;
+
+/// Error type returned from `NormalInverseGaussian::new`
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// `alpha <= 0` or `nan`.
+    AlphaNegativeOrNull,
+    /// `|beta| >= alpha` or `nan`.
+    AbsoluteBetaLessThanAlpha,
+}
+
+/// The [normal-inverse Gaussian distribution](https://en.wikipedia.org/wiki/Normal-inverse_Gaussian_distribution)
+#[derive(Debug)]
+pub struct NormalInverseGaussian<N> {
+    alpha: N,
+    beta: N,
+    inverse_gaussian: InverseGaussian<N>,
+}
+
+impl<N: Float> NormalInverseGaussian<N>
+where StandardNormal: Distribution<N>
+{
+    /// Construct a new `NormalInverseGaussian` distribution with the given alpha (tail heaviness) and
+    /// beta (asymmetry) parameters.
+    pub fn new(alpha: N, beta: N) -> Result<NormalInverseGaussian<N>, Error> {
+        if !(alpha > N::from(0.0)) {
+            return Err(Error::AlphaNegativeOrNull);
+        }
+
+        if !(beta.abs() < alpha) {
+            return Err(Error::AbsoluteBetaLessThanAlpha);
+        }
+
+        let gamma = (alpha.powf(N::from(2.)) - beta.powf(N::from(2.))).sqrt();
```

Comment:
Float multiplication is generally faster than using exponentials.

Reply:
Ok I changed it